src/Domain/PasswordResetToken.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace BigGive\Identity\Domain;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Uid\Uuid;
+
+/**
+ * @ORM\Entity(repositoryClass="BigGive\Identity\Repository\PasswordResetTokenRepository")
+ * @ORM\Table(name="PasswordResetToken", indexes={
+ *     @ORM\Index(name="secret", columns={"secret"}),
+ * })
+ */
+class PasswordResetToken
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="uuid", unique=true)
+     * @ORM\CustomIdGenerator(class="\Symfony\Bridge\Doctrine\IdGenerator\UuidGenerator")
+     */
+    private Uuid $secret;
+
+    /**
+     * @ORM\Column(type="uuid", unique=true)
```

Comment:
If we are going for a new table, I think it would be good to use `@ORM\OneToMany` and its converse so the mapping is enforced and indexed at DBMS level. Or we could simplify and start with tokens just living in `Person` with a requirement that only 1 can be active at a time.

Reply:
Right, I actually hand-wrote the DB migration, didn't remember exactly how to get Doctrine to generate it. Will adjust it to use a foreign key constraint. I think adding the column to the person table would also work, but probably better to allow multiple to be active at the same time to make it easier to use.

Reply:
Using the `vendor/bin/doctrine-migrations diff` command. Currently trying to work out why it wants to drop my created_at and updated_at columns, even if I add the `TimestampsTrait` and `@ORM\HasLifecycleCallbacks` to the `PasswordResetToken` class.

Reply:
I've replaced the person_id property with a Person property. 

I've put in the ManyToOne annotation (many tokens have one person), deleted migration files and regenerated using Doctrine diff. But the generated migration file ( https://github.com/thebiggive/identity/blob/f76d8e740e16bd2d48004a68a044e8b40010f865/src/Migrations/Version20221129154822.php ) still doesn't have a foreign key constraint. Not sure why - https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/association-mapping.html#many-to-one-unidirectional makes it look like it should generate a constraint.